dreamplace/Params.py

```diff
@@ -169,3 +169,70 @@ def solution_file_suffix(self):
             return "def"
         else: # Bookshelf
             return "pl"
+
+def get_bin_size(width, height, num_bins=128*128):
+    """
+    @brief find the power two bin size closest to the canvas ratio.
+    """
+    num_bins_x = math.sqrt(num_bins * width / height)
+    num_bins_x = int(math.pow(2, round(math.log(num_bins_x) / math.log(2))))
+    num_bins_x = max(min(num_bins_x, num_bins), 1) # constrain num_bins_x between 1 and num_bins
+    num_bins_y = int(num_bins / num_bins_x)
+    
+    return num_bins_x, num_bins_y
+
+def get_dreamplace_params(
+    iteration, 
+    target_density,
+    learning_rate,
+    canvas_width=None,
+    canvas_height=None,
+    num_bins_x=None,
+    num_bins_y=None,
+    gpu=False,
+    result_dir='results',
+    legalize_flag=False,
+    stop_overflow=0.1,
+    routability_opt_flag=False):
+    """"
+    @brief return the parameters to config Dreamplace in circuit training.
+    """
+    params = Params()
+    params.use_dp_for_circuit_training = True
```

Comment:
Since you refer ```dp``` to DREAMPlace rather than detailed placement, it is better to change to another name, such as ```circuit_training_mode```

Reply:
Fixed.